src/Serilog.Enrichers.Thread/Enrichers/ThreadIdEnricher.cs

```diff
@@ -43,7 +43,7 @@ public void Enrich(LogEvent logEvent, ILogEventPropertyFactory propertyFactory)
             var threadId = Environment.CurrentManagedThreadId;
 
             var last = _lastValue;
-            if (last == null || (int)((ScalarValue)last.Value).Value != threadId)
+            if (last is null || (int)((ScalarValue)last.Value).Value! != threadId)
```

Comment:
Added the ! for these on the assumption that the ScalarValue is always created here, and doesn't have a null value